web/src/views/InvestigationDetail.vue

```diff
@@ -498,17 +505,84 @@ export default defineComponent({
       }));
     }
 
+    async function toggleNonTiledImageOverlay(pin: Pin, dataset: Dataset) {
+      if (!dataset.content) return;
+      if (!featureLayer) return; // sanity check. this should exist
+      if (!nonTiledOverlayFeature) {
+        nonTiledOverlayFeature = featureLayer.createFeature('quad');
+      }
+      const quadData = nonTiledOverlayFeature.data() || [];
+      // eslint-disable-next-line
```

Comment:
What is the eslint error here?

Reply:
It doesn't like the `any` in `filter((overlay: any), ...)`

Reply:
It might seem silly, but crafting the precise type that your code expects to use here is generally a good idea. Over time, this custom type of `overlay` will grow to reflect what GeoJS expects of it, as your code sees it. It might feel tautological to do that, but there is some benefit that comes from that--that same benefit is being thrown away when you type `overlay` and `any` and ask the typechecker to look the other way.

It's sort of like fastening your seatbelt every time you drive a car, even if it's an edge case like, you're driving from one end of the parking lot to the other.

The analogy is not perfect because seatbelt fastening is quite low cost, and the potential benefits of that cost are quite high. But my larger point is, as you get better at using TypeScript, you can drive the cost of this kind of custom typing down (by developing good habits and practices) and you will accrue small amounts of value over time by doing so.

As an exercise, I'd like you to see how this kind of typing could be done here, with the aim of eliminating the eslint-disable statement (and thereby leaning into typechecking support).

Reply:
Yes, that looks great!